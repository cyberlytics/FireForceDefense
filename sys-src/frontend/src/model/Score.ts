enum Score {
    UNLOCKED,
    ONE_STAR,
    TWO_STARS,
    THREE_STARS,
    LOCKED,
}
export default Score;
