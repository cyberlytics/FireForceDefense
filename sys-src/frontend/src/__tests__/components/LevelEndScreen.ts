import { mount, RouterLinkStub } from '@vue/test-utils';
import LevelEndScreen from '@components/levelEndScreen.vue';
import Score from '@model/Score';

describe('LevelEndScreen.vue', () => {
    test('level end screen (success) matches snapshot', () => {
        const wrapper = mount(LevelEndScreen, {
            propsData: {
                score: Score.THREE_STARS,
                nextLevel: 'lvl002',
            },
            mocks: {
                $t: (key: string) => key,
            },
            stubs: {
                RouterLink: RouterLinkStub,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    test('level end screen (failure) matches snapshot', () => {
        const wrapper = mount(LevelEndScreen, {
            propsData: {
                score: Score.UNLOCKED,
                nextLevel: 'lvl002',
            },
            mocks: {
                $t: (key: string) => key,
            },
            stubs: {
                RouterLink: RouterLinkStub,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    test('level end screen (success, last level) matches snapshot', () => {
        const wrapper = mount(LevelEndScreen, {
            propsData: {
                score: Score.THREE_STARS,
                nextLevel: null,
            },
            mocks: {
                $t: (key: string) => key,
            },
            stubs: {
                RouterLink: RouterLinkStub,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
