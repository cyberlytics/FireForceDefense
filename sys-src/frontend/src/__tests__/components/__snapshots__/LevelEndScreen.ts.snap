// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelEndScreen.vue level end screen (failure) matches snapshot 1`] = `
<div
  class="modal"
  data-backdrop="static"
  data-keyboard="false"
  id="level-end-screen"
>
  <div
    class="modal-dialog modal-dialog-centered"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h2
          class="modal-title h5"
        >
          lost
        </h2>
      </div>
       
      <div
        class="modal-body"
      >
        <div
          class="text-center"
        >
          <img
            alt="0 Sterne"
            class="mb-3"
            src="/assets/levelend_stars_0.svg"
            style="height: 5.5rem;"
          />
        </div>
         
        <!---->
         
        <button
          aria-label="Close"
          class="btn btn-primary btn-block btn-lg"
          data-dismiss="modal"
          data-toggle="modal"
        >
          
                    new start
                
        </button>
         
        <hr
          class="my-3"
        />
         
        <button
          class="btn btn-primary btn-block btn-lg"
          data-dismiss="modal"
        >
          
                    back to level selection
                
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`LevelEndScreen.vue level end screen (success) matches snapshot 1`] = `
<div
  class="modal"
  data-backdrop="static"
  data-keyboard="false"
  id="level-end-screen"
>
  <div
    class="modal-dialog modal-dialog-centered"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h2
          class="modal-title h5"
        >
          victory
        </h2>
      </div>
       
      <div
        class="modal-body"
      >
        <div
          class="text-center"
        >
          <img
            alt="3 Sterne"
            class="mb-3"
            src="/assets/levelend_stars_3.svg"
            style="height: 5.5rem;"
          />
        </div>
         
        <button
          aria-label="Close"
          class="btn btn-primary btn-block btn-lg mb-3"
          data-dismiss="modal"
          data-toggle="modal"
        >
          
                    next level
                
        </button>
         
        <button
          aria-label="Close"
          class="btn btn-primary btn-block"
          data-dismiss="modal"
          data-toggle="modal"
        >
          
                    new start
                
        </button>
         
        <hr
          class="my-3"
        />
         
        <button
          class="btn btn-primary btn-block btn-lg"
          data-dismiss="modal"
        >
          
                    back to level selection
                
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`LevelEndScreen.vue level end screen (success, last level) matches snapshot 1`] = `
<div
  class="modal"
  data-backdrop="static"
  data-keyboard="false"
  id="level-end-screen"
>
  <div
    class="modal-dialog modal-dialog-centered"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h2
          class="modal-title h5"
        >
          victory
        </h2>
      </div>
       
      <div
        class="modal-body"
      >
        <div
          class="text-center"
        >
          <img
            alt="3 Sterne"
            class="mb-3"
            src="/assets/levelend_stars_3.svg"
            style="height: 5.5rem;"
          />
        </div>
         
        <!---->
         
        <button
          aria-label="Close"
          class="btn btn-primary btn-block"
          data-dismiss="modal"
          data-toggle="modal"
        >
          
                    new start
                
        </button>
         
        <hr
          class="my-3"
        />
         
        <button
          class="btn btn-primary btn-block btn-lg"
          data-dismiss="modal"
        >
          
                    back to level selection
                
        </button>
      </div>
    </div>
  </div>
</div>
`;
